### PR TITLE
[parser] enable dynamic import by default

### LINKED ISSUE #10824: Enable dynamic import by default in @babel/parser
## Feature Request

Dynamic `import()` was approved for Stage 4 in June: this means that it should be enabled by default in `@babel/parser` :tada:

It should be done similarly to this commit: https://github.com/babel/babel/commit/24c4901ff56c57db1ac60e4d810424656d18e374

1. Remove all the `this.hasPlugin` and `this.expectPlugin` checks related to this plugin in `packages/babel-parser/src`
2. Remove all the usages of `dynamicImport` in `@babel/parser`'s tests
  1. Probably there is a test to ensure that the plugin is required; it can be removed.
3. Move the tests from the `experimental` folder to `es2020`
6. PR! :tada:

---

If it is the first time that you contribute to Babel, follow these steps: (you need to have `make` and `yarn` available on your machine)
1. **Write a comment there** to let other possible contributors know that you are working on this bug.
1. Fork the repo
1. Run `git clone https://github.com/<YOUR_USERNAME>/babel.git && cd babel`
1. Run `yarn && make bootstrap`
1. Wait :hourglass_flowing_sand: 
1. Run `make watch` (or `make build` whenever you change a file)
1. Add a test if needed (only `input.js`; `output.js` will be automatically generated)
1. Update the code!
1. `yarn jest babel-parser` to run the tests
    - If some test outputs don't match but the new results are correct, you can delete the bad `output.js` files and run the tests again
    - If you prefer, you can run `OVERWRITE=true yarn jest babel-parser` and they will be automatically updated.
1. If it is working, run `make test` to run all the tests
1. Run `git push` and open a PR!

### PR DESCRIPTION
**Pull request by AbdulAli19**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #10824  
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | No
| Documentation PR Link    | 
| Any Dependency Changes?  | No
| License                  | MIT

### CODE REVIEW

**Comment by AbdulAli19** [comment]
couldn't get tests to pass locally so i figure i'm missing something. thought i'd create this pr anyway to get feedback/help on what that might be.

**Comment by nicolo-ribaudo** [comment]
The `babel-preset-env/` failing test is not caused by this PR.
Also, the `/es2020/dynamic-import/no-plugin/input.js` file can be deleted.

**Comment by AbdulAli19** [comment]
> Also, the /es2020/dynamic-import/no-plugin/input.js file can be deleted.

can i also get rid of `/experimental/_no-plugin/dynamic-import/input.js`?

**Comment by AbdulAli19** [comment]
the `babel-preset-env/` test is the only failing test now. thanks for the help, @nicolo-ribaudo!

File: packages/babel-core/test/api.js [hunk 1]
```diff
@@ -754,23 +754,6 @@ describe("api", function() {
       babelrc: false,
     };
 
-    it("only syntax plugin available", function(done) {
```
Comment (nicolo-ribaudo) [inline packages/babel-core/test/api.js]: Instead of removing this test, can you replace it with something else? For example, the pipeline operator.

**Comment by nicolo-ribaudo** [comment]
Rebasing on master should make the tests pass

**Pull request merged**